### PR TITLE
Signer Feature: Transaction Request Modal (with success and error handling)

### LINKED ISSUE #505: Signer Feature: Transaction request sign modal

### PR DESCRIPTION
**Pull request by GuillemGarciaDev**
# Signer Feature: Transaction Request Modal (with success and error handling)

## Issues

closes #505 #507 #508 

## Dependencies

#517 

## Changes

- TransactionRequestModal (modal for signing/rejecting transaction requests) 
- useSignTransactionRequests
- useRejectTransactionRequests
- TransactionRequestScreen
- SignRequestAppSummary (refactored to use it in both modals)
- SignTransactionRequestSuccess + styles
- CopyButton
- locale + testing

### CODE REVIEW

File: src/module/activity/component/display/SignRequestAppSummary/SignRequestAppSummary.tsx [hunk 1]
```diff
@@ -13,40 +14,43 @@ interface SignInRequestSummaryProps {
...
             <Col gap={24} alignItems="center">
-                <Typography variant="body1Strong" textAlign="center">
-                    {translate("confirmConnectionWith")}
+                <Typography variant="body2Strong" textAlign="center">
+                    {requestTitle}
                 </Typography>
                 <DAppImage source={{ uri: image || config.defaultDAppImage }} />
                 <Col alignItems="center">
-                    <Typography variant="body2Strong" textAlign="center">
+                    <Typography variant="body3Strong" textAlign="center">
```
Comment (JordiParraCrespo) [inline src/module/activity/component/display/SignRequestAppSummary/SignRequestAppSummary.tsx:36]: El name igual tmb ha de portar un `numerOfLines={1}` no?

**Review by JordiParraCrespo — CHANGES_REQUESTED** [review]

File: src/locale/locales/el/common.json [hunk 2]
```diff
@@ -203,5 +203,17 @@
...
-    "signInRequestSuccess": "Sign in request signed successfully!"
+    "signInRequestSuccess": "Sign in request signed successfully!",
+    "confirmTransaction": "Confirm transaction",
+    "transaction": "transaction",
+    "seeInExplorer": "See in Explorer",
+    "signTransactionRequest": "Sign transaction request",
+    "signedSuccess": "Signed!",
+    "rejectTransaction": "Reject transaction",
+    "rejectTransactionDescription": "Are you sure you want to reject this transaction? This action cannot be undone.",
+    "rejectConnectionDescription": "Are you sure you want to reject this connection? This action cannot be undone.",
```
[the remark this answers is not in the document]
Comment (GuillemGarciaDev) [reply src/locale/locales/el/common.json:212]: Noo, es diferent

File: src/module/common/component/feedback/SignRequestModal/SignRequestModal.tsx [hunk 3]
```diff
@@ -42,10 +43,12 @@ export default function SignRequestModal({
             })}
             <ConfirmPinModal open={showConfirmation} onClose={() => setShowConfirmation(false)} onConfirmedExited={handleSignRequest} />
             <LoadingModal
+                open={false}
```
[the remark this answers is not in the document]
Comment (GuillemGarciaDev) [reply src/module/common/component/feedback/SignRequestModal/SignRequestModal.tsx:46]: True, però no estava posada a false per defecte

File: src/module/common/component/feedback/CopyButton/CopyButton.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import { useToast } from "@peersyst/react-native-components";
+import * as Clipboard from "expo-clipboard";
+
+export interface CopyButtonProps extends Omit<ButtonProps, "children" | "leftIcon" | "onPress" | "loading"> {
+    copyText: string;
+    showToast?: boolean;
+    toastMessage?: string;
+}
+
+export default function CopyButton({ copyText, showToast: toast = false, toastMessage, ...buttonProps }: CopyButtonProps): JSX.Element {
```
[the remark this answers is not in the document]
Comment (GuillemGarciaDev) [reply src/module/common/component/feedback/CopyButton/CopyButton.tsx:16]: `useCopyToClipboard` de la lib utilitzat al `CopyButton`

File: src/module/common/component/feedback/CopyButton/CopyButton.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    const translate = useTranslate();
+
+    const copyToClipboard = useCopyToClipboard();
+
+    const [copied, setCopied] = useState(false);
+
+    const handleCopy = () => {
+        copyToClipboard({ text: copyText, showToastOnCopy: toast, toastMessage });
+        setCopied(true);
```
Comment (JordiParraCrespo) [inline src/module/common/component/feedback/CopyButton/CopyButton.tsx:24]: Ficali un timeout a false :P

**Review by JordiParraCrespo — APPROVED** [review]

**Pull request merged**